### PR TITLE
snap: prevent duplicated snap name and snap files when parsing seed.yaml

### PR DESCRIPTION
**Pull request by stolowski**
When working on a spread test using seeding for an unrelated PR, I made silly mistakes in my `seed.yaml` and it turned out we don't prevet duplicated entires when parsing (same snap name declared twice, or same snap file referenced twice); this leads to confusing errors later on, so I think it's a good idea to prevent them as soon as possible.

### CODE REVIEW

File: snap/seed_yaml.go [hunk 1]
```diff
@@ -89,6 +91,16 @@ func ReadSeedYaml(fn string) (*Seed, error) {
 		if strings.Contains(sn.File, "/") {
 			return nil, fmt.Errorf("%s: %q must be a filename, not a path", errPrefix, sn.File)
 		}
+
+		// make sure names and file names are unique
```
Comment (bboozzoo) [inline snap/seed_yaml.go]: Should we make sure snap IDs explicitly listed in seed.yaml are unqiue too?

File: snap/seed_yaml.go [hunk 2]
```diff
@@ -89,6 +91,16 @@ func ReadSeedYaml(fn string) (*Seed, error) {
...
+		if seenNames[sn.Name] {
+			return nil, fmt.Errorf("%s: snap name %q not unique", errPrefix, sn.Name)
```
Comment (bboozzoo) [inline snap/seed_yaml.go]: Maybe `name %q must be unique`

File: snap/seed_yaml.go [hunk 3]
```diff
@@ -89,6 +91,16 @@ func ReadSeedYaml(fn string) (*Seed, error) {
...
+		}
+		seenNames[sn.Name] = true
+		if seenFiles[sn.File] {
+			return nil, fmt.Errorf("%s: snap file %q for snap %q not unique", errPrefix, sn.File, sn.Name)
```
Comment (bboozzoo) [inline snap/seed_yaml.go]: I'd go with `snap file %q must be unique`

File: snap/seed_yaml.go (hunk 1 above)
Comment (stolowski) [reply snap/seed_yaml.go]: Good idea, added.

File: snap/seed_yaml.go [hunk 4]
```diff
@@ -89,6 +92,20 @@ func ReadSeedYaml(fn string) (*Seed, error) {
...
+		// make sure names, snap IDs and file names are unique
+		if sn.SnapID != "" && seenSnapIDs[sn.SnapID] {
+			return nil, fmt.Errorf("%s: snap-id %q must be unique", errPrefix, sn.SnapID)
+		}
+		seenSnapIDs[sn.SnapID] = true
+		if seenNames[sn.Name] {
+			return nil, fmt.Errorf("%s: snap name %q must be unique", errPrefix, sn.Name)
+		}
+		seenNames[sn.Name] = true
+		if seenFiles[sn.File] {
```
Comment (mvo5) [inline snap/seed_yaml.go]: Silly question - will this prevent us from installing instances in seed.yaml? I.e. something like:
```
snaps:
 - name: foo
   file: foo.snap
 - name: foo_instance
   file: foo.snap
```
or is this not supported already anyway?

Comment (stolowski) [reply snap/seed_yaml.go]: Very good point, I missed that! I don't think we prevent this (or should prevent). I've reverted this change (as well checking for unique snap ids) for now as they need to be smarter if we want this check - a possible followup. So for now this PR just checks unique names.

File: snap/seed_yaml.go (hunk 1 above)
Comment (zyga) [reply snap/seed_yaml.go]: Enforcement of uniqueness of snap IDs will affect ability to install instances.

File: snap/seed_yaml.go (hunk 4 above)
Comment (pedronis) [reply snap/seed_yaml.go]: we shouldn't support installing instances ATM

**Pull request merged**